client_handlers.go

```diff
@@ -171,7 +172,7 @@ func clientHandshakeHandler(c *Conn) (*alert, error) {
 			if alertPtr, err := handleSingleHandshake(expectedMessages[0].data); err != nil {
 				return alertPtr, err
 			}
-			c.state.localSequenceNumber++
+			c.handshakeMessageSequence++
```

Comment:
I think we can easily get this out of the global state, could help readability a bit I think.

we can put `handshakeMessageSequence` on the stack, and then look at the local cache for the last flight. Or just look at what features are enabled

Reply:
Don't worry about doing this in this PR, merge away!